src/launcher/java/org/truffleruby/launcher/RubyLauncher.java

```diff
@@ -224,6 +224,11 @@ protected void printHelp(OptionCategory maxCategory) {
         printHelp(System.out);
     }
 
+    @Override
+    protected AbortException abortUnrecognizedArgument(String argument) {
+        throw abortInvalidArgument(argument, "truffleruby: invalid option " + argument + "  (Use --help for usage instructions.");
```

Comment:
Should it `throw` and not return the exception?

Reply:
The method it overrides throws as well.